File: tdd-examples_linters/python/test_fizzbuzz.py
from fizzbuzz import fizzbuzz


def test_with_zero():
    assert fizzbuzz(0) == 0


def test_same_number_with_1_2_and_4():
    assert fizzbuzz(1) == 1
    assert fizzbuzz(2) == 2
    assert fizzbuzz(4) == 4


def test_print_Fizz_when_multiple_of_3_only():
    assert fizzbuzz(3) == "Fizz"
    assert fizzbuzz(15) != "Fizz"
    assert fizzbuzz(18) == "Fizz"


def test_print_Fizz_when_multiple_of_5_only():
    assert fizzbuzz(5) == "Buzz"
    assert fizzbuzz(15) != "Buzz"
    assert fizzbuzz(20) == "Buzz"


def test_print_Fizz_when_multiple_of_3_and_5():
    assert fizzbuzz(15) == "FizzBuzz"

File: tdd-examples_linters/ruby/test-fizzbuzz.rb
# Test suite for FizzBuzz
require_relative 'fizzbuzz'
require 'minitest/autorun'

# Test suite for FizzBuzz
class TestFizzBuzz < Minitest::Test
  def test_with_zero
    assert_equal 0, fizzbuzz(0)
  end

  def test_same_number_with_one_two_and_four
    assert_equal 1, fizzbuzz(1)
    assert_equal 2, fizzbuzz(2)
    assert_equal 4, fizzbuzz(4)
  end

  def test_fizz
    assert_equal 'Fizz', fizzbuzz(3)
    assert_equal 'Fizz', fizzbuzz(18)
  end

  def test_buzz
    assert_equal 'Buzz', fizzbuzz(5)
    assert_equal 'Buzz', fizzbuzz(20)
  end

  def test_fizzbuzz
    assert_equal 'FizzBuzz', fizzbuzz(15)
  end
end
